dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,658 @@
+        partial_fit will be called on the model until
+        ``meta['iterations'] >= meta['partial_fit_calls']``
+    X, y : np.ndarray, np.ndarray
+        Training data
+    meta : dict
+        If present, replace ``model_and_meta[1]`` with this object
+    fit_params : dict
+        Keyword args to pass to partial_fit
+
+    Returns
+    -------
+    model : any
+        The model that has been fit.
+    meta : dict
+        A new dictionary with updated information.
+
+    This function does not modify any item in place.
+
+    """
+    start = time()
+    model = deepcopy(model_and_meta[0])
+    if meta is None:
+        meta = deepcopy(model_and_meta[1])
+    else:
+        meta = deepcopy(meta)
+    meta["mean_copy_time"] += time() - start
+    while meta["iterations"] < meta["partial_fit_calls"]:
+        model.partial_fit(X, y, **fit_params)
+        meta["iterations"] += 1
+    meta["mean_fit_time"] += time() - start
+    return model, meta
+
+
+def _score(model_and_meta, x, y, scorer=None, start=0):
+    model, meta = model_and_meta
+    score = scorer(model, x, y)
+
+    score_start = time()
+    meta = deepcopy(meta)
+    meta["mean_copy_time"] += time() - score_start
+    meta.update(score=score)
+    assert meta["iterations"] > 0
+    meta["mean_score_time"] += time() - score_start
+    meta["mean_test_score"] = score
+    meta["time_score_evaluated"] = time() - start
+    return meta
+
+
+def _to_promote(result, completed_jobs, eta=None, asynchronous=True):
+    bracket_models = [
+        r
+        for r in completed_jobs
+        if r["bracket_iter"] == result["bracket_iter"] and
+        r["bracket"] == result["bracket"]
+    ]
+    to_keep = len(bracket_models) // eta
+
+    bracket_completed = len(bracket_models) == result["num_models"]
+    if not asynchronous and not bracket_completed:
+        to_keep = 0
+    if to_keep <= 1:
+        return []
+
+    def _promote_job(job):
+        job["num_models"] = to_keep
+        job["partial_fit_calls"] *= eta
+        job["bracket_iter"] += 1
+        return job
+
+    top = toolz.topk(to_keep, bracket_models, key="score")
+    if not asynchronous:
+        for job in top:
+            job = _promote_job(job)
+        return top
+    else:
+        if to_keep == 0:
+            return [result]
+        if result in top:
+            result = _promote_job(result)
+            return [result]
+        return []
+
+
+def _create_model(model, params, random_state=42):
+    model = clone(model).set_params(**params)
+    if "random_state" in model.get_params():
+        model.set_params(random_state=random_state)
+    return model, None  # right now no meta information
+
+
+def _model_id(s, n_i):
+    return "bracket={s}-{n_i}".format(s=s, n_i=n_i)
+
+
+@gen.coroutine
+def _hyperband(
+    model,
+    params,
+    X,
+    y,
+    max_iter=None,
+    eta=None,
+    fit_params={},
+    random_state=42,
+    test_size=None,
+    scorer=None,
+    asynchronous=None,
```

Comment:
This would be best to set to True or False now.

If we want to change the behavior in the future, we'll want to use `None` as a default to warn with.

Reply:
I might change this in the future, though I don't have any plans to right now. If we change it we can issue a deprecation warning.